aptos-move/aptos-gas/src/transaction/mod.rs

```diff
@@ -222,14 +227,21 @@ impl TransactionGasParameters {
         &self,
         events: impl IntoIterator<Item = &'a ContractEvent>,
     ) -> Fee {
-        events.into_iter().fold(Fee::zero(), |acc, event| {
-            acc + self.storage_fee_per_event_byte * NumBytes::new(event.size() as u64)
-        })
+        let total_bytes = events.into_iter().fold(NumBytes::zero(), |total, event| {
+            total + NumBytes::new(event.size() as u64)
+        });
+        total_bytes
+            .checked_sub(self.free_event_bytes_quota)
+            .unwrap_or(NumBytes::zero())
+            * self.storage_fee_per_event_byte
     }
 
     /// New formula to charge storage fee for transaction based on fixed APT costs.
     pub fn calculate_transaction_storage_fee(&self, txn_size: NumBytes) -> Fee {
-        self.storage_fee_per_transaction_byte * txn_size
+        txn_size
+            .checked_sub(self.large_transaction_cutoff)
```

Comment:
Why are we enforcing large_transaction_cutoff now?

Reply:
To make smaller txns storage fee -free (virtually entire txn is close to free when the gas price is low). Again this is for the new "storage fee", although in this case `large_transaction_cutoff` has been respected for calculating the intrinsic gas https://github.com/aptos-labs/aptos-core/blob/bcdb4f2c98d448bce87151c97803ab7937f61ce0/aptos-move/aptos-gas/src/transaction/mod.rs#L251